src/main/java/org/apache/sling/commons/threads/impl/ThreadPoolExecutorCleaningThreadLocals.java

```diff
@@ -68,8 +69,14 @@ protected void afterExecute(Runnable r, Throwable t) {
 
         if (cleaner != null) {
             cleaner.cleanup();
+            this.cleanupCounter++;
```

Comment:
This is called unconditionally, that means it will increment the counter even if not cleanup was required. I would suggest (SLING-11149 is totally clear in this case) to just count the number of occurrences when a cleanup was actually doing work and releasing ThreadLocals. Then this counter is a useful metric, because any value != 0 will hint to a problem in cleanup (on an application side).

Reply:
@rombert confirmed, that we should only increase the counter when an actual cleanup cleanup was performed and a threadlocal removed.